src/crystal/system/unix/process.cr

```diff
@@ -134,16 +138,28 @@ struct Crystal::System::Process
     end
 
     writer_pipe.close
-    bytes = uninitialized UInt8[4]
-    if reader_pipe.read(bytes.to_slice) == 4
-      message_size = IO::ByteFormat::SystemEndian.decode(Int32, bytes.to_slice)
-      if message_size > 0
-        message = String.build(message_size) { |io| IO.copy(reader_pipe, io, message_size) }
+    begin
+      case reader_pipe.read_byte
+      when nil
+        # Pipe was closed, no error
+      when 0
+        # Error message coming
+        message_size = reader_pipe.read_bytes(Int32)
+        if message_size > 0
+          message = String.build(message_size) { |io| IO.copy(reader_pipe, io, message_size) }
+        end
+        reader_pipe.close
+        raise RuntimeError.new("Error executing process: '#{command_args[0]}': #{message}")
```

Comment:
On Windows it prints the full command, maybe here it should too?

Reply:
I don't know... most errors happens because of the command, not the args and I don't want to clutter the output with all the arguments. Actually I think the abstraction layer should have the same interface and receive the arguments separately.

Reply:
> most errors

Umm I think non-typical errors also deserve to be debuggable. Also not sure I agree with the prediction; I'd say malformed args are more widespread, and it's unfair/confusing to say that an executable failed without any context, when it's the args that caused it.

> don't want to clutter the output with all the arguments

> arguments separately

The process with the arguments are both inherent parts of the command line (see: `cat /proc/$pid/cmdline`), I think we can't just say that one of them is less important.

Reply:
Ah you know what, maybe we *can* say that the args are less important, but I still want to be able to debug what's happening.

Reply:
No strong feelings about the abstraction, but I also wouldn't say that trying to use it in this way shows that the abstraction is wrong.

Reply:
If there's an issue with the args the process is still executed. I don't know if there is any corner case, but it doesn't raise an exception because the process is created successfully. In any case, the process will return a non-zero exit code if there is an issue with the arguments.
I just tested in Ruby, Python and Java. All of them raises exceptions with just the process name, not the args. I think we should fix the Windows implementation to do the same.